### PR TITLE
Check User Profile permissions before loading settings page

### PR DESCRIPTION
**Pull request by davidjray**
# Critical Changes

# Changes
- Only allows User Profiles with the "Customize Application" setting to access the HEDA settings page.

# Issues Closed
- Implements #270

### CODE REVIEW

**Comment by davidjray** [comment]
image

File: src/classes/UTIL_CustomSettingsFacade.cls [hunk 1]
```diff
@@ -50,6 +52,16 @@ public without sharing class UTIL_CustomSettingsFacade {
     static List<Relationship_Lookup__c> relationshipLookup;
 
     /*******************************************************************************************************
+    * @description Constructor, sets boolean value to determine is User can access Settings page 
+    * 
+    */    
+    public UTIL_CustomSettingsFacade() {
+        User user = [SELECT ProfileId FROM User where Id = :UserInfo.getUserId()]; 
+        Profile profile = [SELECT PermissionsCustomizeApplication FROM Profile Where Id = :user.ProfileId];
```
Comment (ceiroa) [inline src/classes/UTIL_CustomSettingsFacade.cls]: Can we combine these two queries into one?

Comment (ceiroa) [inline src/classes/UTIL_CustomSettingsFacade.cls]: Also, shouldn't this be the pull request in which we create a separate class to be the component's controller and move those methods that don't belong out of UTIL_CustomSettingsFacade?

File: src/classes/UTIL_CustomSettingsFacade.cls [hunk 2]
```diff
@@ -37,6 +37,8 @@
 */
 public without sharing class UTIL_CustomSettingsFacade {
 
+    public Boolean hasPerm {get;set;}
```
Comment (ceiroa) [inline src/classes/UTIL_CustomSettingsFacade.cls]: Does this need to be public for the component to be able to read it? If not, better make it private.

Comment (davidjray) [inline src/classes/UTIL_CustomSettingsFacade.cls]: It does need to be public since the VF page uses it.

**Comment by ceiroa** [comment]
@davidjray Also ready to be merged?

**Comment by davidjray** [comment]
@ceiroa, yes ready to be merged.

**Pull request merged**